### ISSUE TITLE
Pie-in-the-sky RFE: automatic wordclouds for each meeting

### ISSUE THREAD

**Issue by mattdm** [fedora-infra/mote#92]
Remy and Ralph hooked up this cool thing on twitter (https://twitter.com/fedobot/status/707270047803121664) which automatically generates wordclouds from each meeting. It'd be awesome to have this integrated into Mote directly.

And as an enhancement, I'd like to see _two_ clouds per meeting — one for usernames (including usernames mentioned in chat) and one for everything else.(Maybe use a different color scheme for each.)

**Comment by cydrobolt** [comment]
This seems like a useful nifty feature we could definitely integrate. I can look around for generators of my own, but if you already have one that is working well, I can take a look at it and see if it can be integrated :)

**Comment by mattdm** [comment]
@decause @ralphbean I have no idea how the current thing is hooked up. :)

**Comment by ralphbean** [comment]
I think [this](https://github.com/decause/wordcloudbot) is the code.  @decause, can you confirm?

**Comment by decause** [comment]
@ralphbean yes, can confirm this is the code. The PoC is hosted locally on my personal machine (not ideal) but would like to be deployed somewhere more permanent, but there are some issues:

1) It chokes on auto-generated tweets larger than 140 chars, and needs to fail gracefully: https://github.com/decause/wordcloudbot/issues/2
2) If we are going to deploy this within Fedora Infrastructure, then word_cloud would need to be packaged for Fedora.
3) If we are going to deploy it outside of Fedora Infrastructure, then we need a place to put it.

HTH,
--RemyD.

**Comment by cydrobolt** [comment]
I might be down to package `word_cloud` in Fedora. We could also try using JavaScript on the client side to create word clouds.

https://github.com/jasondavies/d3-cloud

**Comment by jflory7** [comment]
As an update, the code for `wordclodubot` is on Pagure: [pagure.io/wordcloudbot](https://pagure.io/wordcloudbot "systemd service that listens to fedmsg for IRC meeting logs, generates a word cloud, and tweets it to a Twitter account")

**Comment by mattdm** [comment]
Here's the [most recent FESCo meeting](https://meetbot.fedoraproject.org/fedora-meeting/2017-07-21/fesco.2017-07-21-16.00.log.html) using the javascript-based thing. For the names, I just took the 2nd column for the name list, and removed zodbot and `*` (smarter thing would be to take _3rd_ column when a `*` appears). For the text, I took third column on. It'd be better to also completely remove names from this list, but I didn't here. Anyway, they do look nice!

![fesco-people](https://user-images.githubusercontent.com/123367/28484130-bc1e3e7e-6e3e-11e7-8585-f9afcc3cee04.png)
![fesco-words](https://user-images.githubusercontent.com/123367/28484131-bc234cac-6e3e-11e7-97b7-d42d93bf0ff8.png)

**Comment by mattdm** [comment]
Latest Fedora Magazine article, column 3 only:

![magazine](https://user-images.githubusercontent.com/123367/28484328-ac6976e6-6e3f-11e7-9c23-cdeb07179f21.png)
